### DISCUSSION TITLE
Reverse-alias with custom domain

### DISCUSSION THREAD

**Discussion by Vnzq4Qje3zzU** [simple-login/app#827] [Feature Requests]
The idea would be to change the format "parsedContactAddress_randomNoise@simplelogin.co" to "parsedContactAddress_myAlias@mydomain.tld".

Advantages:

1. Allow sending from reverse-alias without creating a contact manually (similar to the so wanted way AnonAddy uses [1]): eliminates the [eager desire for SMTP](https://github.com/simple-login/app/discussions/679) temporarily.
2. Similar format to DuckDuckGo's privacy email: will help people coming from there to understand how it works (they probably have tons of users already). [2]
3. Removes burden from simplelogin.co (I think). I have read that one of the reasons for using simplelogin.co for reverse-alias was deliverability of the domain, but that isn't really necessary if you own the domain (no decent email service will classify the emails as spam if you are receiving from that domain constantly).

[1] AnonAddy's reverse alias for `abc123+xyz456@example.com` would be `myAlias+abc123+xyz456=example.com@mydomain.tld`, which isn't close to what SL or DDG is doing. The problem is with alias with '+' and the forced use of special characters. https://anonaddy.com/help/sending-email-from-an-alias/

[2] Unfortunately, the `parsedContactAddress` part is not parsed in the exactly same way as DDG. `abc123+xyz456@example.com` becomes `abc123_xyz456.at.example.com` with SL while DDG would parse as `abc123+xyz456_at_example.com`. [Because you can't have an underscore in the domain part](https://en.wikipedia.org/wiki/Email_address#Domain) and '+' signs are kept, the DDG's way is strictly better at avoiding ambiguities than SL's approach.

**Comment by Vnzq4Qje3zzU** [comment]
Adopting DDG's `parsedContactAddress` format means that '+' won't be replaced by '_', which means that [reverse alias replacement may not work as intended](https://github.com/simple-login/app/issues/418#issuecomment-865834025) - but, I was testing, and the bug seems to only happen on Gmail when there is more than one plus sign, which I think is a small sacrifice comparing to all advantages (also, random reverse-alias is available anyway).

**Comment by nguyenkims** [comment]
With other SL users, we have actually discussed a lot about using a formula for creating reverse alias. The TLDR is I don't think this is a good idea because:
- harder anti-spoofing. Email spoofing is a tough topic and there's basically no authentication on the MTA level. Using a predictable reverse alias only makes this problem harder to solve.
- not easy to use by non-tech users
- easier to be abused: as SL has a good email sending reputation, a lot of people want to use the service to send out marketing emails (that we are obviously against). With a fixed formula for creating reverse aliases, it's easy for bulk emailing.

What's really missing here IMO is an easy way to create reverse alias which can be achieved with client apps like browser extension or mobile app.

**Comment by nguyenkims** [comment]
About using a custom domain for reverse alias, this can be indeed a good idea. However there are some advantages of sending emails from simplelogin.co domain:
- avoid the headaches of handling conflicts between reverse alias and alias on a custom domain. For example, let's say a reverse alias sender_example_com@domain.com is created and a user wants to create an alias with the same address on their domain. This scenario is unlikely but can happen in theory.
- simplelogin.co domain reputation becomes stronger and stronger over time which benefits other SL users as well :).
